src/code/code_80140E80.c

```diff
@@ -1,21 +1,359 @@
+    bg->b.frameW = width << 2;
+    bg->b.imageY = 0;
+    bg->b.imageH = height << 2;
+    bg->b.frameY = y * 4;
+    bg->b.frameH = height << 2;
+    bg->b.imageLoad = G_BGLT_LOADTILE;
+    bg->b.imageFmt = G_IM_FMT_RGBA;
+    bg->b.imageSiz = G_IM_SIZ_16b;
+    bg->b.imagePal = 0;
+    bg->b.imageFlip = 0;
+    bg->b.imagePtr = source;
+
+    if (!!(cycleFlag & BG_CYC_COPY) != 0) { //! TODO possibly fake, may be a better way
+        guS2DInitBg(bg);
+    } else {
+        bg->s.scaleW = (s32)((1 << 10) / scaleX);
+        bg->s.scaleH = (s32)((1 << 10) / scaleY);
+        bg->s.imageYorig = bg->b.imageY;
+    }
+
+    // draw BG to `img`
+    func_80140EAC(&gfx, bg, img, width, height, cycleFlag);
+
+    *gfxP = gfx;
+}
+
+// used in FbdemoWipe5 and internally
+// "default settings" wrapper for func_80141008
+/**
+ * Set up a BG from a specified source image and draw it to the specified color image with func_80140EAC(), using the
+ * BG's settings. Position uses the default (0,0), and no rescaling is done even 1-cycle mode is enabled.
+ *
+ * @see func_80141008() for arguments.
+ */
+void func_8014116C(Gfx** gfxP, void* source, void* img, s32 width, s32 height, s32 cycleFlag) {
+    func_80141008(gfxP, source, img, width, height, 0.0f, 0.0f, SCALE_UNCHANGED, SCALE_UNCHANGED, cycleFlag);
+}
+
+// wrapper for func_80141008 with general arguments, used in func_80141200
+/**
+ * Set up a BG from a specified source image and draw it to the specified color image with func_80140EAC(). Fully
+ * general settings are available.
+ *
+ * @see func_80141008() for arguments.
+ */
+void func_801411B4(Gfx** gfxP, void* source, void* img, s32 width, s32 height, f32 x, f32 y, f32 scaleX, f32 scaleY,
+                   s32 cycleFlag) {
+    func_80141008(gfxP, source, img, width, height, x, y, scaleX, scaleY, cycleFlag);
+}
+
+// TODO work out what to do with this.
+#define gDPSetPrimColor_u32(pkt, m, l, d)                                                      \
+    _DW({                                                                                      \
+        Gfx* _g = (Gfx*)(pkt);                                                                 \
+                                                                                               \
+        _g->words.w0 = (_SHIFTL(G_SETPRIMCOLOR, 24, 8) | _SHIFTL(m, 8, 8) | _SHIFTL(l, 0, 8)); \
+        _g->words.w1 = (unsigned int)(d);                                                      \
+    })
+
+#define gDPSetLODColor(pkt, c, m, l, d)                                          \
+    _DW({                                                                        \
+        Gfx* _g = (Gfx*)(pkt);                                                   \
+                                                                                 \
+        _g->words.w0 = _SHIFTL(c, 24, 8) | _SHIFTL(m, 8, 8) | _SHIFTL(l, 0, 8)); \
+        _g->words.w1 = (unsigned int)(d);                                        \
+    })
```

Comment:
More gfx macros with alt definitions? Just like that prerender one. Great...

Reply:
This one isn't too bad, though, it's just doing a texture load from a `Color_RGBA8_u32`. But yeah, be nice to not have to worry about it...

Reply:
I guess we should put it on macros.h for now

Reply:
Any reason not to just use `gDPSetLODColor` with `G_SETPRIMCOLOR`, instead of making `gDPSetPrimColor_u32` as well.

Reply:
The question here is whether it's
- a generalisation of `gDPSetPrimColor` to allow for a word instead of individual colours
- a generalisation of `gDPSetColor` to allow for specifying LOD as well
and which of these it is would determine the macro to use. Notably `gDPSetEnvColor` is a special case of `gDPSetColor` but `gDPSetPrimColor` is defined from scratch

Reply:
I don't think it really matters which way, though I think my preference is using `gDPSetLODColor` since we already have an instance of `gDPSetColor` used with `G_SETPRIMCOLOR` when setting a word is needed in `gfxprint`. Though if we do go with generalizing `gDPSetPrimColor` might I suggest `gDPSetPrimColorWord` as a name.
Also agree with Anghelo and it should be in macros.h, at least for now.